### ISSUE TITLE
OpenSearch 2.19 validation

### ISSUE THREAD

**Issue by mako42** [Graylog2/graylog2-server#21025]
2.19.3 will be released by July 22th, 2025 and therefore will be a suitable candidate to be included in Data Node. 

- [x] run backend integration tests
- [x] run e2e tests onPrem
- [x] run e2e tests in the cloud
- [x] install OS 2.19.3 on https://graylog-test70
- [x] test upgrade path (from last released data node version) Docker 
- [x] test multinode Data Node setup upgrade (Docker)

**Comment by user29835461** [comment]
What is the status this in relation to the upcoming 6.3 release? OpenSearch 2.15 which is as per document the highest supported version has not received critical security patches for a long time now.

**Comment by drewmiranda-gl** [comment]
Hi @user29835461 ,
We plan to upgrade support to a later version of OpenSearch with Graylog 7.0.

Graylog 7.0 is scheduled for General Availability release in November 2025.

**Comment by logopk** [comment]
That's quite an unfortunate decision.

OS 2.15 is now 1yr old and itself and the included java version  is 21.0.4 have known vulnerabilities...

**Comment by user29835461** [comment]
> OS 2.15 is now 1yr old and itself and the included java version is 21.0.4 have known vulnerabilities...

I guess they don't have any customers with regulatory requirements for vulnerability management. There is no way a such customer can use this software.

**Comment by ed-ud** [comment]
This is really important, and I know OpenSearch is a "moving target" and doesn't have LTS releases, they don't do "bugfix releases", etc.  But that's the platform Graylog has chosen to run on...

Graylog needs to support the latest 2.x version... and it won't be long until 3.x support is really needed when 2.x becomes EOL.

As mentioned in this and other threads, we can't be expected to continue with extremely outdated versions of OpenSearch (2.15 is old), and all of the security bugs that aren't addressed... not to mention all the components that come with OpenSearch (Java, node, etc.) that also have outstanding security bugs.

I thought we were done with this after the move to OpenSearch (and getting away from ES 7.10.x and running it for years past EOL with no security fixes).

**Comment by kroepke** [comment]
I agree that the situation is unfortunate, and I share the sentiment that this particular dependency has caused some bumps in the road.

While I don't want to make up excuses or argue against any of the points raised here, I do want to give a little context:
* We have reviewed the medium to high severity CVEs in 2.15, and none directly affect our usage or normal usage when deployed in a Graylog cluster. So yes, the report doesn't look great, but when doing the due diligence, it's not nearly as bad as it seems at first glance.
* It is certainly possible to run Opensearch with a different JDK than the one the distribution ships. If you are concerned about that particular Adoptium version, please consider running a different one.
* 2.16, 2.17, and 2.18 all had issues that prevented us from shipping them in a release. If we could've easily updated, we would've, but we spent considerable time working with upstream to isolate the issues. The fact of the matter is that 2.19 is the first version that we could've shipped, but we wanted more testing.
* And then we ran out of time.
 
Again, I won't make up excuses or gaslight you into "this is good". It's not, and your criticism is heard clearly and much appreciated.

**Comment by HenryTheSir** [comment]
Hi @kroepke 

if I understand you correctly 2.19 is supported to use but not shipped with datanode? Can you change your docs according? 

Thanks!


ref https://github.com/Graylog2/graylog2-server/issues/22662

**Comment by mako42** [comment]
At this point, we also don't officially support 2.19. We have seen issues in 2.19.0 causing problems for some users (depends on the setup and used feature set; for some use cases it's absolutely fine). We hope that these issues are fixed with 2.19.3 and therefore started testing this version. Once we know for sure, we will adjust our docs accordingly.

**Comment by logopk** [comment]
Hi, as 2.19.3 is available, what would be the best guess when you could give us a go? Thank You.

**Comment by mako42** [comment]
Hi @logopk. The evaluation of 2.19.3 is already ongoing and looks good so far, but we still need to confirm that the issues from past versions are fixed. 
Once we're done, we will let you know. If all goes well, we should be able to give the green light in a week, maybe two.

**Comment by ed-ud** [comment]
Great news!  We are running 5.1.13 and were planning to upgrade to 6.x
soon, but then found this situation with OpenSearch compatibility concerns.
Reloading all of our index data from tape backups is not a very appealing
prospect.

Graylog won't work without OpenSearch - and they release new versions
rather willy-nilly... so that means Graylog needs to be "recertified" with
each new
OpenSearch release in a reasonable time frame to avoid known security
problems with all the components OpenSearch brings with it (that have
security issues of their own).
Not ideal, but since Graylog chose OpenSearch for indexing, that's the
reality of depending on OpenSearch.

-- Ed

**Comment by ed-ud** [comment]
How goes the evaluation of 2.19.3 with Graylog?

**Comment by mako42** [comment]
Good news! We've just finished the evaluation successfully - all known compatibility issues have been fixed and we did not find any new ones. We'll update our documentation and compatibility matrix shortly and will bump the version in data node with our upcoming release. Thank you for your patience!

**Comment by ed-ud** [comment]
Hello,

I noticed the validation has been done for 6.2 and 6.3 according to:
https://go2docs.graylog.org/current/downloading_and_installing_graylog/compatibility_matrix.htm#CompatibilityforGraylogOpenandEnterprisewithOpenSearch

Should we expect troubles as we upgrade from 5.x to 6.0 in preparation to
upgrade to 6.3, given that we are already running OpenSearch 2.19.3 with
Graylog 5.1.13?

Thanks,
-- Ed

Ed Phillips ***@***.***> University of Delaware x6082
Systems Programmer IV, IT-ESCS UNIX Systems Engineering

**Comment by moesterheld** [comment]
@ed-ud 
No, there should be no problems updating to 6.x/7.0 when you are already running OS 2.19.3

There were issues in 2.18 and 2.19 prior to 2.19.2 which prevented us from supporting it.

**Comment by Som-gif** [comment]
Hello everyone,

We are currently considering an upgrade of our Wazuh Indexer from version 4.9.2 (OpenSearch 2.13) to 4.12 (OpenSearch 2.19.1).

From what I understand, OpenSearch 2.19.3 is fully compatible with Graylog 6.3, while OpenSearch 2.19.1 may have some known issues. Could you please provide details on the specific problems identified with OpenSearch 2.19.1? This would help us determine whether these issues could affect our environment and whether it is safe for us to proceed with the Wazuh Indexer upgrade.

Thank you in advance for your help.

**Comment by mako42** [comment]
We ran into [this issue ](https://github.com/opensearch-project/OpenSearch/issues/17518) in 2.19.1, not sure how this affects Wazuh. 
Since we are done with the validation for us, I will close the issue.
